docs/CrossCompilationModel.md

```diff
@@ -0,0 +1,103 @@
+
+# Cross-Compilation Model
+
+## Components
+
+When compiling Swift code, the compiler will consult three different sources of
+inputs outside of the user provided code.
+
+1. Compiler Resources
+2. System (C/C++) Headers (Platform SDK)
+3. Platform Swift Modules (Swift SDK)
+
+These pieces compose in a particular manner to build a system image to build
+code against.
+
+The compiler resources are content shipped with the toolchain and are tied to
+that specific version and build of the compiler. In the case of the Swift
+compiler, this includes the Swift shims. Whilst this is a compiler resource, the
+packaging may not necessarily be part of the toolchain due to interdependencies.
+Some of this content is required to process the system headers themselves (e.g.
+clang's builtin resource headers).
+
+The C/C++ system headers (and libraries) are what is traditionally called the
+Unix "sysroot". On most Unix systems, this is the set of headers that are
+associated with the system library set which normally may be found in
+`/usr/include`. On Darwin, this is included in the unified SDK which is shipped
+to the user, while on Windows, this is called the Windows SDK, which is a
+separate installable component. For simplicity, we will refer to this as the
+Platform SDK universally.
+
+The platform Swift modules provided by the Swift SDK is unified with the
```

Comment:
Maybe it's just me, but my brain is having a hard time parsing this sentence.

> The platform Swift modules provided by the Swift SDK is unified with the Platform SDK.

Is this that the module maps for the platform are shipped as part of the Swift SDK? I think that is already covered by

>  It contains the libraries and overlays that provide the core language runtime and expose system libraries to Swift code in an ergonomic manner. This may be in the form of API Notes, module maps, wrapper types, or Swift `extension`s, or entire libraries.

I would like to see Swift SDK defined before use though.
Maybe something more like this?

> The Swift SDK contains libraries and overlays that provide the core language
runtime and expose system libraries to Swift code in an ergonomic manner. This
may be in the form of API Notes, module maps, wrapper types, Swift
`extension`s to platform types, or entire overlay libraries. In some distributions, the Swift SDK and
platform SDK are unified, such as the SDKs for Apple platforms, where the C, C++,
and Swift libraries and interfaces are shipped together.

Thoughts?

Reply:
Adjusted as discussed offline.